### PR TITLE
04.Version cache - debug framework

### PR DESCRIPTION
**Pull request by Kalimuthu-Velappan**
Added DBGOPT Make variable to enable the cache framework
scripts in trace mode. This option takes the part name of the script to
enable the particular shell script in trace mode.

Multiple shell script names can also be given.

        Eg: make DBGOPT="image|docker"

Added verbose mode to dump the version merge details during
build/dry-run mode.
        Eg: scripts/versions_manager.py freeze -v \
                'dryrun|cmod=docker-swss|cfile=versions-deb|cname=all|stage=sub|stage=add'

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Comment by adyeung** [comment]
@xumia @liushilongbuaa submitter has taken the time to split the original code PR to smaller submissions for review, pls help take a look

**Comment by zhangyanzhao** [comment]
@xumia @liushilongbuaa can you please review the PR? Thanks.

File: Makefile.work [hunk 1]
```diff
@@ -221,12 +234,14 @@ SLAVE_TAG = $(shell \
 	(cat $(SLAVE_DIR)/Dockerfile.user \
 		 $(SLAVE_DIR)/Dockerfile \
 		 $(SLAVE_DIR)/buildinfo/versions/versions-* \
-		 .git/HEAD \
+		.git/HEAD 2>/dev/null \
```
Comment (xumia) [inline Makefile.work:237]: Can we remove the .git/HEAD? It has impact on PR check.

**Comment by Kalimuthu-Velappan** [comment]
It is merged as part of 
https://github.com/sonic-net/sonic-buildimage/pull/12005

**Pull request closed without merging**